### PR TITLE
feat(TDP-1676): extract month day labels

### PR DESCRIPTION
**Pull request by tlnd-cgall**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-1676

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/40/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/40/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/41/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/41/)

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java [hunk 1]
```diff
@@ -115,15 +132,31 @@
         parameters.add(parameter(locale).setName(YEAR).setType(BOOLEAN).setDefaultValue(TRUE).build(this));
         parameters.add(parameter(locale).setName(QUARTER).setType(BOOLEAN).setDefaultValue(FALSE).build(this));
         parameters.add(parameter(locale).setName(MONTH).setType(BOOLEAN).setDefaultValue(TRUE).build(this));
+        parameters.add(parameter(locale).setName(MONTH_LABEL).setType(BOOLEAN).setDefaultValue(TRUE).build(this));
```
Comment (smallet) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java]: Should be unchecked by default

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java [hunk 2]
```diff
@@ -77,12 +85,19 @@
...
+
     /** Day of year constant value. */
     private static final String DAY_OF_YEAR = "DAY_OF_YEAR";
 
     /** Week of year constant value. */
     private static final String WEEK_OF_YEAR = "WEEK_OF_YEAR";
 
+    private static final String LANGUAGE = "LANGUAGE";
+
+    private static String LOCALE;
```
Comment (smallet) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java]: Is not used in the code

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java [hunk 3]
```diff
@@ -77,12 +85,19 @@
...
+
     /** True constant value. */
     private static final String TRUE = "true";
```
Comment (smallet) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java]: Better to use Boolean.xxx instead
Not introduced by this PR.

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java [hunk 4]
```diff
@@ -196,6 +241,32 @@ int getQuarter(int numMonth) {
...
+        }
+    }
+
+    /**
+     * Get the month label of the year.
+     *
+     * @param numMonth the number of the month
+     * @return the month label
+     */
+    String getLabelMonth(int numMonth, Locale locale) {
```
Comment (smallet) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java]: Could be nice to add some unit tests on this atomic function

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java [hunk 5]
```diff
@@ -196,6 +241,32 @@ int getQuarter(int numMonth) {
         return (numMonth - 1) / 3 + 1;
     }
 
+    /**
+     * Get the day label of the week.
+     *
+     * @param numDayOfWeek the number of the month
+     * @return the day label
+     */
+    String getLabelDay(int numDayOfWeek, Locale locale) {
```
Comment (smallet) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokens.java]: Could be nice to add some unit tests on this atomic function

File: dataprep-actions/src/test/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokensTest.java [hunk 6]
```diff
@@ -265,13 +301,36 @@ public void test_getQuarter() {
         assertEquals(1, action.getQuarter(1));
     }
 
+    @Test
+    public void test_getLabelDay() {
+        Locale locale = Locale.US;
```
Comment (smallet) [inline dataprep-actions/src/test/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokensTest.java]: it seems it's not used

File: dataprep-actions/src/test/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokensTest.java [hunk 7]
```diff
@@ -265,13 +301,36 @@ public void test_getQuarter() {
...
+        assertThat(action.getLabelDay(1, Locale.US), is("Monday"));
+        assertThat(action.getLabelDay(3, Locale.US), is("Wednesday"));
+        assertThat(action.getLabelDay(7, Locale.US), is("Sunday"));
+    }
```
Comment (smallet) [inline dataprep-actions/src/test/java/org/talend/dataprep/transformation/actions/date/ExtractDateTokensTest.java]: add tests with invalid values (70, -7 for example)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/60/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/60/)

**Pull request merged**